otter/test_files/ok_test.py

```diff
@@ -49,9 +49,19 @@ def run_doctest(name, doctest_string, global_environment):
         result = doctestrunner.summarize(verbose=True)
     # An individual test can only pass or fail
     if result.failed == 0:
-        return (True, '')
+        return True, ''
     else:
-        return False, runresults.getvalue()
+        from ..assign.assignment import Assignment
+        if Assignment().traceback_length == 'full':
+            return False, runresults.getvalue()
+        elif Assignment().traceback_length == 'assertion_msg':
+            err_msg = runresults.getvalue()
+            if 'AssertionError: ' in err_msg:
+                return False, err_msg[err_msg.index('AssertionError: '):]
+            else:
+                return False, ''
+        elif Assignment().traceback_length == 'none':
+            return False, ''
```

Comment:
Calling `Assignment()` just creates an `Assignment` instance with the default values, so `traceback_length` will always be the default.